File: grind75/Dec22/150.py
import math
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        operators = '+-*/'
        for token in tokens:
            if token not in operators:
                stack.append(int(token))
            else:
                operand2 = stack.pop()
                operand1 = stack.pop()
                if token == '+':
                    stack.append(operand1 + operand2)
                elif token == '-':
                    stack.append(operand1 - operand2)
                elif token == '/':
                    stack.append(math.trunc(operand1 / operand2))
                elif token == '*':
                    stack.append(operand1 * operand2)
        
        return stack[0]

File: grind75/Dec22/150.go
func evalRPN(tokens []string) int {
    stack := []string{}

    for _, token := range tokens {
        if _, ok := strconv.Atoi(token); ok == nil {
            stack = append(stack, token)
            continue
        }
        if len(stack) < 2 { return -1 }
        l, _ := strconv.Atoi(stack[len(stack)-2])
        r, _ := strconv.Atoi(stack[len(stack)-1])
        stack = stack[:len(stack)-2]
        tmp := 0
        if token == "+" { tmp = l+r }
        if token == "-" { tmp = l-r }
        if token == "*" { tmp = l*r }
        if token == "/" { tmp = l/r }
        stack = append(stack, strconv.Itoa(tmp))
    }
    
    if len(stack) != 1 { return -1 }
    ret, _ := strconv.Atoi(stack[0])
    return ret
}
